src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -1025,8 +1025,8 @@ def main():
         if proc:
             proc.terminate()
 
-    if not keep_job_dir:
-        shutil.rmtree(job_dir)
+        if not keep_job_dir:
+            shutil.rmtree(job_dir)
```

Comment:
when above in try returns, `job_dir` might not be deleted if it's outside of finally.

Reply:
no. in try, function can return. 

```
In [4]: def example_function():
   ...:     try:
   ...:         return
   ...:     finally:
   ...:         print('foo')
   ...: 
   ...:     print('bar')
   ...: 
   ...: example_function()
foo

In [5]:
```

Reply:
yeah, I made a toy example. 👍